sockeye/training.py

```diff
@@ -586,14 +588,12 @@ def fit(self,
                 for name, val in metric_val.get_name_value():
                     logger.info('Checkpoint [%d]\tValidation-%s=%f', self.state.checkpoint, name, val)
 
-                # (3) update training metrics
-                self._update_metrics(metric_train, metric_val, process_manager)
-                metric_train.reset()
-
-                # (4) determine improvement
+                # (2) determine improvement
                 has_improved = False
                 previous_best = self.state.best_metric
-                for checkpoint, metric_dict in enumerate(self.state.metrics, 1):
+                # at this point state.self.metrics doesn't have validation results yet
+                current_checkpoint_val_metric = {"%s-val" % name:val for name, val in metric_val.get_name_value()}
```

Comment:
this will not actually work if the optimized metric is `BLEU` as it will only later be added to the metrics.

Reply:
I now moved the waiting for collect_result() out of the _update_metrics() to let state.metrics have the bleu-val if it's computed. Tested locally.